editor/src/clj/internal/node.clj

```diff
@@ -1544,7 +1544,9 @@
   (node-id [this] node-id)
   (node-type [this] node-type)
   (get-property [this basis property]
-    (get properties property (gt/get-property (gt/node-by-id-at basis original-id) basis property)))
+    (if-some [entry (find properties property)]
+      (val entry)
+      (gt/get-property (gt/node-by-id-at basis original-id) basis property)))
```

Comment:
Evaluating properties on override nodes would always look up every node up the override chain and evaluate the property on each of them unnecessarily. Now we only recurse up the chain when our node doesn't provide an override value.